### ISSUE TITLE
Agent takes longer than running semgrep directly

### ISSUE THREAD

**Issue by brendongo** [returntocorp/sgrep-action#35]
Looks like semgrep-agent is passing files to semgrep using --includes which causes semgrep to traverse the directory tree to see if any files match the include pattern.

Fix will be to pass the files directly to semgrep as a target argument.

**Comment by underyx** [comment]
I remember implementing it like that at first, and it didn't work 😞 

I think the issue was that when targeting specific files, semgrep would just assume that it's the correct language for the given rules, and it would return tons of parse errors (since if an HTML file was changed, it'd get directly targeted, while the repo might configured for `r/python`). I used directory targeting and post-filtering files so that semgrep would stop scanning files of the wrong language.

Another issue I think I recall is that rule IDs were being rewritten with relative path fragments, but my memory is hazy and this one might have been a different problem.

**Comment by brendongo** [comment]
Once https://github.com/returntocorp/semgrep/pull/1507 is merged and released
can update https://github.com/returntocorp/semgrep-action/blob/develop/src/semgrep_agent/semgrep.py#L138 to send targets instead of includes. (Also need to update to use this cli flag)
